clients/auth/auth.go

```diff
@@ -106,86 +143,81 @@ func (c Client) Create(ctx context.Context, params *CreateParams) error {
-			writePerm:    params.WriteDashboardsPermission,
-			resourceType: "dashboards",
-		},
-		{
-			readPerm:     params.ReadNotificationEndpointPermission,
-			writePerm:    params.WriteNotificationEndpointPermission,
-			resourceType: "notificationEndpoints",
-		},
-		{
-			readPerm:     params.ReadNotificationRulePermission,
-			writePerm:    params.WriteNotificationRulePermission,
-			resourceType: "notificationRules",
-		},
-		{
-			readPerm:     params.ReadOrganizationsPermission,
-			writePerm:    params.WriteOrganizationsPermission,
-			resourceType: "orgs",
-		},
-		{
-			readPerm:     params.ReadTasksPermission,
-			writePerm:    params.WriteTasksPermission,
-			resourceType: "tasks",
-		},
-		{
-			readPerm:     params.ReadTelegrafsPermission,
-			writePerm:    params.WriteTelegrafsPermission,
-			resourceType: "telegrafs",
-		},
-		{
-			readPerm:     params.ReadUserPermission,
-			writePerm:    params.WriteUserPermission,
-			resourceType: "users",
-		},
-		{
-			readPerm:     params.ReadDBRPPermission,
-			writePerm:    params.WriteDBRPPermission,
-			resourceType: "dbrp",
-		},
-	}
-
-	for _, provided := range providedPerm {
+	for _, resourcePermission := range params.ResourcePermissions {
 		var actions []string
-		if provided.readPerm {
+		if resourcePermission.Read {
 			actions = append(actions, ReadAction)
 		}
-		if provided.writePerm {
+		if resourcePermission.Write {
 			actions = append(actions, WriteAction)
 		}
 
 		for _, action := range actions {
 			p := api.Permission{
 				Action:   action,
-				Resource: makePermResource(provided.resourceType, "", orgID),
+				Resource: makePermResource(resourcePermission.Name, "", orgID),
 			}
 			permissions = append(permissions, p)
 		}
 	}
 
-	// Get the user ID because the command only takes a username, not ID
-	users, err := c.UsersApi.GetUsers(ctx).Name(params.User).Execute()
-	if err != nil || len(users.GetUsers()) == 0 {
-		return fmt.Errorf("could not find user with name %q: %w", params.User, err)
+	if params.OperatorPermission {
+		if len(permissions) > 0 {
+			return fmt.Errorf("cannot compose other permissions with operator permissions")
+		}
+		resources, err := c.GetResources(ctx).Execute()
+		if err != nil {
+			return err
+		}
+		for _, r := range resources {
+			for _, action := range []string{ReadAction, WriteAction} {
+				permissions = append(permissions, api.Permission{
+					Action:   action,
+					Resource: makePermResource(r, "", ""),
+				})
+			}
+		}
+	}
+
+	if params.AllAccess {
+		if len(permissions) > 0 {
+			return fmt.Errorf("cannot compose other permissions with all access permissions")
+		}
+		resources, err := c.GetResources(ctx).Execute()
+		if err != nil {
+			return err
+		}
+		for _, r := range resources {
+			if r == string(extras.RESOURCEENUMCLOUD_ORGS) {
+				// orgs are handled specifically - all access is for a single org, not global access to all orgs
+				permissions = append(permissions, api.Permission{
+					Action: ReadAction,
```

Comment:
Sanity-checking: should all-access also get `WriteAction` on the org?

Reply:
Specifically no - at least, the ui repo does not do this. See `src/authorizations/utils/permissions.ts` .

```
  if (t === 'orgs') {
    // 'orgs' used to only have read permissions so that's all we'll give again.
    // In production, orgs with an orgID returns a permissions error.
    return [
      {
        action: 'read' as 'read',
        resource: {type: t, id: orgID},
      },
    ]
  }
```